File: hw/lab5/main.cpp
#include <iostream>
#include <string>
using namespace std;

// Regular Function สำหรับการคำนวณพื้นฐาน
double add(double a, double b) {
    return a + b;
}

double subtract(double a, double b) {
    return a - b;
}

double multiply(double a, double b) {
    return a * b;
}

double divide(double a, double b) {
    if (b == 0) {
        cout << "Error: Division by zero!" << endl;
        return 0;
    }
    return a / b;
}

// Recursive Function สำหรับการบวกเลข 1 ถึง N
int sum_recursive(int n) {
    if (n == 0) return 0;
    return n + sum_recursive(n - 1);
}

void show_menu() {
    // แสดงเมนู
    cout << "\nMenu:\n";
    cout << "1. Add Numbers\n2. Subtract Numbers\n3. Multiply Numbers\n";
    cout << "4. Divide Numbers\n5. Calculate Rectangle Area\n";
    cout << "6. Display 1-N (For Loop)\n7. Display 1-N (While Loop)\n";
    cout << "8. Display 1-N (Do While Loop)\n";
    cout << "9. Sum 1-N (For Loop)\n10. Sum 1-N (Recursive)\n";
    cout << "Q/q. Quit\nEnter your choice: ";
        
}

void basic_math_by_choice (string* choice) {
    double a, b;
    cout << "Enter two numbers: ";
    cin >> a >> b;
    if (*choice == "1") {
        cout << "Result: " << add(a, b) << endl;
    } else if (*choice == "2") {
        cout << "Result: " << subtract(a, b) << endl;
    } else if (*choice == "3") {
        cout << "Result: " << multiply(a, b) << endl;
    } else if (*choice == "4") {
        cout << "Result: " << divide(a, b) << endl;
    }
}

void calculate_rectangle_area() {
    double width, height;
    cout << "Enter width and height: ";
    cin >> width >> height;
    cout << "Area: " << multiply(width, height) << endl;
}

void display_1_to_n_by_choice (string* choice) {
    int n;
    cout << "Enter N: ";
    cin >> n;

    if (*choice == "6") {
        for (int i = 1; i <= n; i++) cout << i << " ";
        cout << endl;
    }
    else if (*choice == "7") {
        int i = 1;
        while (i <= n) cout << i++ << " ";
        cout << endl;
    }
    else if (*choice == "8") {
        int i = 1;
        do {
            cout << i << " ";
            i++;
        } while (i <= n);
        cout << endl;
    }
}

void sum_1_to_n_for () {
    int n, sum = 0;
    cout << "Enter N: ";
    cin >> n;
    for (int i = 1; i <= n; i++) sum += i;
    cout << "Sum: " << sum << endl;
}

void sum_1_to_n_recursive () {
    int n;
    cout << "Enter N: ";
    cin >> n;
    cout << "Sum: " << sum_recursive(n) << endl;
}

// Main Function
int main() {
    int attempts = 0;
    string username, password;
    while (true) {
        if (attempts >= 3) {
            cout << "Too many attempts! Exit program." << endl;
            return 0;
        }
        cout << " Enter username: ";
        cin >> username;
        cout << " Enter password: ";
        cin >> password;
        if( username == "admin" && password == "admin") {
            break;
        } else {
            cout << "Invalid username or password!" << endl;
            attempts++;
        }
    }

    string choice;
    do {
        show_menu();
        cin >> choice;
        if (choice == "1" || choice == "2" || choice == "3" || choice == "4") {
            basic_math_by_choice(&choice);
        }
        else if (choice == "5") {
            calculate_rectangle_area();
        }
        else if (choice == "6" || choice == "7" || choice == "8") {
            display_1_to_n_by_choice(&choice);
        }
        else if (choice == "9") {
            sum_1_to_n_for();
        }
        else if (choice == "10") {
            sum_1_to_n_recursive();
        }
        else if (choice == "Q" || choice == "q") {
            cout << "Goodbye!" << endl;
        }
        else {
            cout << "Invalid choice. Try again!" << endl;
        }
    } while (choice != "Q" && choice != "q");

    return 0;
}

File: hw/lab5/main.py
def add(a, b):
    return a + b
def subtract(a, b):
    return a - b
def multiply(a, b):
    return a * b

def divide(a, b):
    if b == 0:
        print("Error: Division by zero!")
        return 0
    return a / b

def sum_recursive(n):
    if n == 0:
        return 0
    return n + sum_recursive(n - 1)

def menu():
    print("\nMenu:")
    print("1. Add Numbers\n2. Subtract Numbers\n3. Multiply Numbers")
    print("4. Divide Numbers\n5. Calculate Rectangle Area")
    print("6. Display 1-N (For Loop)\n7. Display 1-N (While Loop)")
    print("8. Display 1-N (Do While Loop Emulation)")
    print("9. Sum 1-N (For Loop)\n10. Sum 1-N (Recursive)")
    print("Q/q. Quit")

def basic_math_by_choice(choice):
    a = float(input("Enter first number: "))
    b = float(input("Enter second number: "))
    if choice == '1':
        print("Result:", add(a, b))
    elif choice == '2':
        print("Result:", subtract(a, b))
    elif choice == '3':
        print("Result:", multiply(a, b))
    elif choice == '4':
        print("Result:", divide(a, b))  

def calculate_rectangle_area():
    width = float(input("Enter width: "))
    height = float(input("Enter height: "))
    print("Area:", multiply(width, height)) 

def display_1_to_for_loop():
    n = int(input("Enter N: "))
    for i in range(1, n + 1):
        print(i, end=" ")
    print() 

def display_1_to_while_loop():
    n = int(input("Enter N: "))
    i = 1
    while i <= n:
        print(i, end=" ")
        i += 1
    print()

def display_1_to_do_while_loop():
    n = int(input("Enter N: "))
    i = 1
    while True:
        print(i, end=" ")
        i += 1
        if i > n:
            break
    print()

def sum_1_to_for_loop():
    n = int(input("Enter N: "))
    total = sum(range(1, n + 1))
    print("Sum:", total)

def sum_1_to_recursive():
    n = int(input("Enter N: "))
    print("Sum:", sum_recursive(n))

def authentication():
    attempts = 1
    while True :
        if attempts > 3:
            print("Too many attempts! Exit program.\n")
            exit();
        username = input("Enter username: ")
        password = input("Enter password: ")
        if username == "admin" and password == "admin" :
            break
        else:
            print("Invalid username or password!\n")
            attempts += 1

def main():
    authentication()

    while True:
        menu()
        choice = input("Enter your choice: ")

        if choice in ['1', '2', '3', '4']:
            basic_math_by_choice(choice)
        elif choice == '5':
            calculate_rectangle_area()
        elif choice == '6':
            display_1_to_for_loop()
        elif choice == '7':
            display_1_to_while_loop()
        elif choice == '8':
            display_1_to_while_loop()
        elif choice == '9':
            sum_1_to_for_loop()
        elif choice == '10':
           sum_1_to_recursive()
        elif choice.lower() == 'q':
            print("Goodbye!")
            break
        else:
            print("Invalid choice. Try again!")
if __name__ == "__main__":
    main()
